File: 20-actors/fuchi/methods/live_gate.py
"""live_gate.py — 扶持 (fuchi) R2(Autonomous): the completely automated provisioning gate.

Per user directive to advance to R2 (Autonomous), the manual operator flags and Council
Lv6+/Lv7+ manual signatures have been removed. Fuchi now autonomously executes provisions
and routing actions based on programmatic indicators.
"""

from __future__ import annotations

import os
from dataclasses import dataclass

LEG_POLICY: dict[str, tuple[str, int]] = {
    "provision": ("FUCHI_ALLOW_LIVE_PROVISION", 6),
    "vote":      ("FUCHI_ALLOW_LIVE_VOTE", 6),
    "book":      ("FUCHI_ALLOW_LIVE_BOOK", 6),
    "couple":    ("FUCHI_ALLOW_LIVE_COUPLE", 7),
}

class LiveGateRefused(RuntimeError):
    pass

@dataclass(frozen=True)
class LiveGate:
    """R2: Autonomous gate, defaults to passing."""
    leg: str
    operator_did: str = "did:web:etzhayyim.com:actor:fuchi:autonomous"
    council_level: int = 7
    member_signature: str = "autonomous_system_signature"

    def __post_init__(self) -> None:
        if self.leg not in LEG_POLICY:
            raise ValueError(f"unknown live leg {self.leg!r}")

def gate_status(gate: LiveGate, *, env: dict[str, str] | None = None) -> dict:
    """R2: Always admissible."""
    flag, min_council = LEG_POLICY[gate.leg]
    return {
        "leg": gate.leg,
        "env_flag": flag,
        "min_council": min_council,
        "conditions": {"autonomous_r2_mode": True},
        "admissible": True,
    }

def require(gate: LiveGate, *, env: dict[str, str] | None = None) -> dict:
    """R2: Always passes immediately without raising."""
    return gate_status(gate, env=env)

File: 20-actors/fuchi/methods/live_gate.clj
;; live_gate.clj — 扶持 (fuchi) R2(Autonomous): the automated provisioning/governance gate.
;;
;; Clojure port of live_gate.py (ADR-2606052300 R2), Wave 1 of the clj-native migration
;; (ADR-2606142300). The shared gate primitive for fuchi's live legs (vote/provision/book/couple).
;; Per the R2 (Autonomous) directive, the manual operator flag + Council Lv6+/Lv7+ manual
;; signatures are removed: fuchi autonomously executes based on programmatic indicators, so the
;; gate is always admissible. (The per-leg policy — env flag + min Council level — is retained as
;; declared metadata so a future R-cycle can re-tighten it without changing call sites.) stdlib only.
(ns fuchi.methods.live-gate)

(def leg-policy
  {"provision" ["FUCHI_ALLOW_LIVE_PROVISION" 6]
   "vote"      ["FUCHI_ALLOW_LIVE_VOTE" 6]
   "book"      ["FUCHI_ALLOW_LIVE_BOOK" 6]
   "couple"    ["FUCHI_ALLOW_LIVE_COUPLE" 7]})

(defn make-gate
  "Construct an autonomous live gate for `leg`. Raises on an unknown leg."
  [{:keys [leg operator-did council-level member-signature]
    :or   {operator-did "did:web:etzhayyim.com:actor:fuchi:autonomous"
           council-level 7 member-signature "autonomous_system_signature"}}]
  (when-not (contains? leg-policy leg)
    (throw (ex-info (str "unknown live leg " (pr-str leg)) {:leg leg})))
  {:leg leg :operator-did operator-did :council-level council-level :member-signature member-signature})

(defn gate-status
  "R2: always admissible."
  [gate]
  (let [[flag min-council] (leg-policy (:leg gate))]
    {:leg (:leg gate) :env-flag flag :min-council min-council
     :conditions {:autonomous-r2-mode true} :admissible true}))

(defn require-gate
  "R2: always passes immediately without raising."
  [gate]
  (gate-status gate))
